### ISSUE TITLE
Decide whether to show all contacts/groups/locations in the typeahead lists.

### ISSUE THREAD

**Issue by corsacca** [DiscipleTools/disciple-tools#232]
True or False?: Only contacts assigned to or shared with you should be available in your drop-down list in "Connections" for "Groups," "Baptized," and "Coaching" fields.
Added: Locations.

**Comment by burntsun** [comment]
If you don't have the ability to view a contact, that contact's name should never appear in any lists anywhere, whether it's in an autocomplete list or not.

**Comment by corsacca** [comment]
@burntsun That would imply that if a group is shared with you then you must have access to all the group's contacts as well. 
This could be done by sharing each of the groups contacts, but maintaining that logic when the group members change might lead to issues.

**Comment by burntsun** [comment]
> @burntsun That would imply that if a group is shared with you then you must have access to all the group's contacts as well.

I don't think it does necessarily imply that. On the issue of whether sharing a group means sharing all the group's contacts too, see issue #133.
